ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -138,6 +140,11 @@ internal constructor(
         }
   }
 
+  private fun applySelectedState(features: Set<Feature>, selectedLoiId: String?): Set<Feature> =
```

Comment:
Nit feel free to ignore—this name feels a bit general since we only want this for LOIs, but we don't encode that fact in the signature (since we take a string). Maybe a more specific name would help readability ever so slightly? Or are we intending this to be more general in the future (selections working for other features)?

Reply:
Good points. How about this:

* `applySelectedState` -> `updatedLoiSelectedStates`
* Move `selected` from `Style` to `Feature`
* Replace `withSelected(..)` with `select()` and `deselect()` on `Feature`

Reply:
Note that this function also needs to deselect LOIs which are no longer selected.

Reply:
Sounds like a great plan to me!